src/Tabs.js

```diff
@@ -81,7 +82,7 @@ const defaultProps = {
 };
 
 function getDefaultActiveKey(children) {
-  let defaultActiveKey;
+  let defaultActiveKey = null;
```

Comment:
What issue is this fixing?

Reply:
To my knowledge, the default value for a variable declared by `let's unspecified. As a result, the line 87 `if (defaultActiveKey == null)`has unpredictable behavior. This eliminates the uncertainty.

Reply:
and it's very possible that is why it's mandatory to provide a defaultActiveKey property in order to have a tab active by default

Reply:
the default value for an uninitialized variable in JS is perfectly well-defined – it's always `undefined`.

Reply:
then it makes it clear that this fix is necessary for line 87 to match the condition

Reply:
but it’s already clear, and it’s not necessary to have this line for it to be clear...

Reply:
I wrote "this fix is necessary for line 87 to match the condition"
If `defaultActiveKey` is undefined, then `if (defaultActiveKey == null)`never matches, and the method always returns `undefined`.
This is probably the cause of the bug I've observed, where no tab is active unless you specify the defaultActiveKey _property_.

Reply:
null == undefined in JS. it doesn’t pass ===, but the use of == is intentional here

Reply:
then I guess my original comment 'clarify intent' is a good enough reason for this harmless change, coz all this is only clear to a small subset of JS developers...

Reply:
but the main problem is that it’s not adding any *value* here. There’s not really a point in adding this change if there’s nothing significant that is being added here.

> coz all this is only clear to a small subset of JS developers

I believe that’s making an unfair assumption on the people who will read this portion of the code. At most, we could add a comment here.

Reply:
just proposed an entirely new version which is clear, concise and should run faster (iteration stops of first candidate key)

Reply:
> but the main problem is that it’s not adding any _value_ here. There’s not really a point in adding this change if there’s nothing significant that is being added here.
> 
> > coz all this is only clear to a small subset of JS developers
> 
> I believe that’s making an unfair assumption on the people who will read this portion of the code. At most, we could add a comment here.

Sorry to hear that we have such different perceptions of _clarity_ and _value_...

Reply:
> the default value for an uninitialized variable in JS is perfectly well-defined – it's always `undefined`.

Probably true, but can you point to an mdn spec that says this is true for `let`? I couldn't find one, hence my say that it's unspecified.

Reply:
it's right here: https://www.ecma-international.org/ecma-262/6.0/#sec-let-and-const-declarations

> If a LexicalBinding in a let declaration does not have an Initializer the variable is assigned the value undefined when the LexicalBinding is evaluated.